algorithms/src/polycommit/sonic_pc/polynomial.rs

```diff
@@ -33,18 +33,50 @@ use itertools::Itertools;
 #[cfg(feature = "parallel")]
 use rayon::prelude::*;
 
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize, Eq, PartialEq)]
+pub struct PolynomialInfo {
+    label: PolynomialLabel,
+    degree_bound: Option<usize>,
+    hiding_bound: Option<usize>,
+}
+
+impl PolynomialInfo {
+    /// Construct a new labeled polynomial by consuming `polynomial`.
+    pub fn new(label: PolynomialLabel, degree_bound: Option<usize>, hiding_bound: Option<usize>) -> Self {
+        Self { label, degree_bound, hiding_bound }
+    }
+
+    /// Return the label for `self`.
+    pub fn label(&self) -> &String {
```

Comment:
```suggestion
    pub fn label(&self) -> &str {
```
nit: it's better to return `&str` instead of `&String` whenever possible